client/service/src/config.rs

```diff
@@ -191,3 +195,52 @@ impl Default for RpcMethods {
 		RpcMethods::Auto
 	}
 }
+
+/// The base path that is used for everything that needs to be write on disk to run a node.
+pub enum BasePath {
+	/// A temporary directory is used as base path and will be deleted when dropped.
+	#[cfg(not(target_os = "unknown"))]
+	Temporary(TempDir),
+	/// An path existing path on the disk.
+	Permanenent(PathBuf),
+}
+
+impl BasePath {
+	/// Create a temporary directory prefixed with "substrate" and use it as base path.
+	#[cfg(not(target_os = "unknown"))]
+	pub fn new_temp_dir() -> io::Result<BasePath> {
+		Ok(BasePath::Temporary(
+			tempfile::Builder::new().prefix("substrate").tempdir()?,
+		))
+	}
+
+	/// Create a base path based on an existing path on disk.
```

Comment:
I think we create it if it does not exists?

Reply:
Not explicitly. We create only the network_config_dir which is a subdirectory. We could explicitly create the directories. Though this doc sentence actually meant "create a BasePath struct".

Reply:
Than please write it. I did not understand it like that.

Reply:
You're totally right. Done in cb140b6eee63c209339552280a8fb77909cb24fa